messages/java/src/main/java/io/cucumber/messages/NdjsonToMessageIterable.java

```diff
@@ -1,43 +1,53 @@
 package io.cucumber.messages;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.Reader;
 import java.nio.charset.StandardCharsets;
 import java.util.Iterator;
-
-import static io.cucumber.messages.Messages.*;
+import java.util.function.BiFunction;
 
 /**
  * Iterates over messages read from a stream. Client code should not depend on this class
  * directly, but rather on a {@code Iterable<Messages.Envelope>} object.
  * Tests can then use a {@code new ArrayList<Messages.Envelope>} which implements the same interface.
  */
-public final class NdjsonToMessageIterable implements Iterable<Envelope> {
-    private final BufferedReader input;
-    private Envelope next;
+public final class NdjsonToMessageIterable<T> implements Iterable<T>, AutoCloseable {
```

Comment:
I think it's okay to keep this pinned to `Envelope`.

Reply:
In theory yes, but when I was debugging the internals of Jackson trying to figure out why it wouldn't deserialize a single argument constructor (see `handles_single_argument_constructors`) I wanted to have the tiniest object possible.

Reply:
The problem with Jackson has been solved so I've removed it again. This is part of the public API and this would make it more complex the needed.